code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +624,250 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
+		return "Asteroid#Small02";
+	case XWMObjectType::oj_Asteroid3:
+		return "Asteroid#Medium01";
+	case XWMObjectType::oj_Asteroid4:
+		return "Asteroid#Medium02";
+	case XWMObjectType::oj_Asteroid5:
+		return "Asteroid#Medium03";
+	case XWMObjectType::oj_Asteroid6:
+		return "Asteroid#Big01";
+	case XWMObjectType::oj_Asteroid7:
+		return "Asteroid#Big02";
+	case XWMObjectType::oj_Asteroid8:
+		return "Asteroid#Big03";
+	default:
+		break;
+	}
+	return nullptr;
+}
+
+vec3d xwi_determine_mine_formation_position(const XWMObject* oj, float objectPosX, float objectPosY, float objectPosZ, 
+	float offsetAxisA, float offsetAxisB)
+{
+	switch (oj->formation) {  // Y and Z axes must be switched for FSO
+	case XWMObjectFormation::ojf_FloorXY:
+		return vm_vec_new((objectPosX + offsetAxisA), objectPosZ, (objectPosY + offsetAxisB));
+	case XWMObjectFormation::ojf_SideYZ:
+		return vm_vec_new(objectPosX, (objectPosZ + offsetAxisA), (objectPosY + offsetAxisB));
+	case XWMObjectFormation::ojf_FrontXZ:
+		return vm_vec_new((objectPosX + offsetAxisA), (objectPosZ + offsetAxisB), objectPosY);
+	case XWMObjectFormation::ojf_Scattered:
+		return vm_vec_new(objectPosX, objectPosZ, objectPosY);
+	default:
+		break;
+	}
+	return vm_vec_new(objectPosX, objectPosZ, objectPosY);
+}
+
+void xwi_determine_object_orient(matrix* orient, const XWMObject* oj)
+{
+	angles a;
+	a.p = oj->object_pitch;
+	a.b = oj->object_roll;
+	a.h = oj->object_yaw;
+	vm_angles_2_matrix(orient, &a);
+	return;
+}
+
+// Determine the unique name from the object comprised of the object type and suffix. 
+// Add the new name to the objectNameSet and return it to parse_xwi_objectgroup
+const char* xwi_determine_space_object_name(SCP_set<SCP_string>& objectNameSet, const char* object_type)
+{
+	char base_name[NAME_LENGTH];
+	char suffix[NAME_LENGTH];
+	strcpy_s(base_name, object_type);
+	end_string_at_first_hash_symbol(base_name);
+
+	// we'll need to try suffixes starting at 1 and going until we find a unique name
+	int n = 1;
+	char object_name[NAME_LENGTH];
+	do {
+		sprintf(suffix, NOX(" %d"), n++);
+
+		// start building name
+		strcpy_s(object_name, base_name);
+
+		// if generated name will be longer than allowable name, truncate the class section of the name by the overflow
+		int char_overflow = static_cast<int>(strlen(base_name) + strlen(suffix)) - (NAME_LENGTH - 1);
+		if (char_overflow > 0) {
+			object_name[strlen(base_name) - static_cast<size_t>(char_overflow)] = '\0';
+		}
+
+		// complete building the name by adding suffix number and converting case
+		strcat_s(object_name, suffix);
+		SCP_totitle(object_name);
+
+		// continue as long as we find the name in our set
+	} while (objectNameSet.find(object_name) != objectNameSet.end());
+
+	// name does not yet exist in the set, so it's valid; add it and return
+	auto iter = objectNameSet.insert(object_name);
+	return iter.first->c_str();
+}
+
+void parse_xwi_objectgroup(mission* pm, const XWingMission* xwim, const XWMObject* oj)
```

Comment:
Search the file for `// load flight groups` and add a loop for the object groups similar to the existing loop for flight groups